Tutorials/Tutorial27_PostProcessing/src/Tutorial27_PostProcessing.cpp

```diff
@@ -0,0 +1,615 @@
+{
+
+    auto pShaderSourceFactory = CreateCompoundShaderSourceFactory(pDevice);
+
+    ShaderCreateInfo ShaderCI;
+    ShaderCI.EntryPoint                      = EntryPoint;
+    ShaderCI.FilePath                        = FileName;
+    ShaderCI.Macros                          = Macros;
+    ShaderCI.SourceLanguage                  = SHADER_SOURCE_LANGUAGE_HLSL;
+    ShaderCI.Desc.ShaderType                 = Type;
+    ShaderCI.Desc.Name                       = EntryPoint;
+    ShaderCI.pShaderSourceStreamFactory      = pShaderSourceFactory;
+    ShaderCI.Desc.UseCombinedTextureSamplers = true;
+    return RenderDeviceWithCache<false>{pDevice, pStateCache}.CreateShader(ShaderCI);
+}
+
+static DILIGENT_CONSTEXPR DepthStencilStateDesc DSS_DepthWrite{
+    True,                   // DepthEnable
+    True,                   // DepthWriteEnable
+    COMPARISON_FUNC_ALWAYS, // DepthFunc
+};
+
+namespace HLSL
+{
+
+#include "Shaders/Common/public/BasicStructures.fxh"
+#include "Shaders/PostProcess/ToneMapping/public/ToneMappingStructures.fxh"
+#include "Shaders/PostProcess/ScreenSpaceReflection/public/ScreenSpaceReflectionStructures.fxh"
+
+} // namespace HLSL
+
+enum GBUFFER_RT : Uint32
+{
+    GBUFFER_RT_BASE_COLOR,
+    GBUFFER_RT_MATERIAL_DATA,
+    GBUFFER_RT_NORMAL,
+    GBUFFER_RT_OBJECT_IDENTIFIER,
+    GBUFFER_RT_DEPTH,
+    GBUFFER_RT_COUNT
+};
+
+enum GBUFFER_RT_FLAG : Uint32
+{
+    GBUFFER_RT_FLAG_BASE_COLOR     = 1u << GBUFFER_RT_BASE_COLOR,
+    GBUFFER_RT_FLAG_MATERIAL_DATA  = 1u << GBUFFER_RT_MATERIAL_DATA,
+    GBUFFER_RT_FLAG_NORMAL         = 1u << GBUFFER_RT_NORMAL,
+    GBUFFER_RT_FLAG_MOTION_VECTORS = 1u << GBUFFER_RT_OBJECT_IDENTIFIER,
+    GBUFFER_RT_FLAG_DEPTH          = 1u << GBUFFER_RT_DEPTH,
+    GBUFFER_RT_FLAG_LAST           = GBUFFER_RT_FLAG_DEPTH,
+    GBUFFER_RT_FLAG_ALL            = (GBUFFER_RT_FLAG_LAST << 1u) - 1u,
+};
+DEFINE_FLAG_ENUM_OPERATORS(GBUFFER_RT_FLAG);
+
+SampleBase* CreateSample()
+{
+    return new Tutorial27_PostProcessing();
+}
+
+struct Tutorial27_PostProcessing::ShaderParams
+{
+    HLSL::ToneMappingAttribs           ToneMapSettings = {};
+    HLSL::ScreenSpaceReflectionAttribs SSRSettings     = {};
+
+    float IBLScale        = 1.0;
+    float OcclusionScale  = 1.0;
+    float ReflectionScale = 1.0;
+};
+
+Tutorial27_PostProcessing::Tutorial27_PostProcessing() :
+    m_Resources{RESOURCE_IDENTIFIER_COUNT},
+    m_CameraAttribs{std::make_unique<HLSL::CameraAttribs[]>(2)}
+{
+    m_Camera.SetMoveSpeed(4.0f);
+    m_Camera.SetPos(float3(-6.5f, 1.5f, 6.5f));
+    m_Camera.SetReferenceAxes(float3(1.0f, 0.0f, 0.0f), float3(0.0f, 1.0f, 0.0f), false);
+    m_Camera.SetLookAt(float3(0.0f, 0.0f, 0.0f));
+
+
+    m_ObjectTransforms.resize(m_MaxObjectCount, float4x4::Identity());
+}
+
+void Tutorial27_PostProcessing::Initialize(const SampleInitInfo& InitInfo)
+{
+    SampleBase::Initialize(InitInfo);
+
+    /// Creating a layout for G-Buffe
```

Comment:
Typo: Create a layout for G-Buffer
Also, use normal comments `//`, not the doxygen comments `///`